File: LAB02/Palindromo.java
public class Palindromo {
    public static void main(String[] args) {
        
        String palavra;
        int contador;
        palavra = MyIO.readLine();
        int tamanho;
        if(!palavra.equals("FIM")){
            do{
                contador = 0;
                tamanho = palavra.length();
                for(int i = 0;i < tamanho;i++){ 
                    if(palavra.charAt(i) == palavra.charAt(tamanho - 1 - i)){
                        contador++;
                    }
                }
                if(contador == tamanho){
                    MyIO.print("SIM\n");
                }else{
                    MyIO.print("NAO\n");
                }
                palavra = MyIO.readLine();
            }while(!palavra.equals("FIM"));
        }
    }
}

File: tps/TP 01/Q02 - Palindromo/palindromo.c
#include <stdio.h>
#include <string.h>

char palavra[150];

void leitura(){
    scanf(" %[^\n]", palavra);
}

void palindromo(){
    
    int tamanho = strlen(palavra);
    int contador = 0;
    
    for(int i = 0; i < tamanho; i ++){
        
        if(palavra[i] == palavra[tamanho-i-1]){
            contador++;
        }
    }
    if(contador == tamanho){
        printf("SIM\n", palavra);
    }
    else{
        printf("NAO\n", palavra);
    }
}

int main()
{
    leitura();
    
    while(strcmp(palavra,"FIM")!= 0){
        palindromo();
        leitura();
    }

    return 0;
}
